m4/aclocal_fc.f90

```diff
@@ -203,24 +206,29 @@ PROGRAM FC08_AVAIL_KINDS
 
       num_rkinds = SIZE(real_kinds)
 
+      ! some compilers (ACfL 24) reported real=16 kind, but refused to
+      ! compile with it.  Verify the kind can be selected in SELECTED_REAL_KIND.
+      found_rkinds(:) = .false.
       max_decimal_prec = 1
 
       prec: DO ik = 2, 36
          exp: DO jk = 1, 700
             k = SELECTED_REAL_KIND(ik,jk)
             IF(k.LT.0) EXIT exp
+            do kk = 1,num_rkinds
+               if (real_kinds(kk) == k) found_rkinds(kk) = .true.
+            end do
             max_decimal_prec = ik
```

Comment:
If the KIND  is not supported, then the maximum decimal precision should not be set for that KIND.